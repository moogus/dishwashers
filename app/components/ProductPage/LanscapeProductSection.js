import React from 'react';
import { string } from 'prop-types';

import ProductImage from 'components/ProductImage';
import ProductDescription from 'components/ProductDescription';
import ProductPrice from 'components/ProductPrice';

import {
  H2,
  H3,
  TextMarginContainer,
  GridContainer,
  LanscapeGridInnerContainer,
} from './styled';

const LandscapeProductSection = ({ code }) => (
        <div className="d-none d-lg-block -xl-block">
          <GridContainer>
            <LanscapeGridInnerContainer>
              <div className="row">
                <div className="col-8  pr-0">
                  <ProductImage />
                  <TextMarginContainer className="row">
                    <div className="col">
                      <H2>Product Information</H2>
                    </div>
                  </TextMarginContainer>
                  <TextMarginContainer className="row">
                    <div className="col ">
                      <H3>{`Product code: ${code}`}</H3>
                    </div>
                  </TextMarginContainer>
                  <TextMarginContainer className="row">
                    <ProductDescription/>
                  </TextMarginContainer>
                </div>
                <div className="col-4 border-left">
                  <ProductPrice/>
                </div>
              </div>
            </LanscapeGridInnerContainer>
          </GridContainer>
        </div>
);

LandscapeProductSection.propTypes = { code: string };

export default LandscapeProductSection;
