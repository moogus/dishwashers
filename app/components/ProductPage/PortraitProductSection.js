import React from 'react';
import { string } from 'prop-types';

import ProductImage from 'components/ProductImage';
import ProductDescription from 'components/ProductDescription';
import ProductPrice from 'components/ProductPrice';

import {
  H2,
  H3,
  TextMarginContainer,
  GridContainer,
  LineSpacer,
  PortraitGridInnerContainer,
} from './styled';

const PortraitProductSection = ({ code }) => (
  <div className="d-none d-sm-block d-md-block d-lg-none">
    <GridContainer className="row">
      <PortraitGridInnerContainer className="col">
        <ProductImage />
        <LineSpacer className="row pt1 pb1 border-top border-bottom"/>
        <ProductPrice/>
        <TextMarginContainer className="row">
          <div className="col">
            <H2>Product Information</H2>
          </div>
        </TextMarginContainer>
        <TextMarginContainer className="row">
          <div className="col">
            <H3>{`Product code: ${code}`}</H3>
          </div>
        </TextMarginContainer>
        <TextMarginContainer className="row">
          <ProductDescription/>
        </TextMarginContainer>
      </PortraitGridInnerContainer>
    </GridContainer>
  </div>
);

PortraitProductSection.propTypes = { code: string };

export default PortraitProductSection;
