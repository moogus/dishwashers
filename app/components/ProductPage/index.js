import React from 'react';
import { string } from 'prop-types';
import { useNavigate } from '@reach/router';

import PortraitProductPage from 'components/ProductPage/PortraitProductSection';
import LandscapeProductPage from 'components/ProductPage/LanscapeProductSection';

import useLoadProduct from 'hooks/product/load-product';
import useGetProduct from 'hooks/product/get-product';

import {
  H1,
  NavigateBackButton,
  Loader,
  PageContainer,
} from './styled';

const ProductPage = ({ productId }) => {
  const { loading } = useLoadProduct(productId);
  const { title, code } = useGetProduct();
  const navigate = useNavigate();

  return loading ? (
    <Loader>Loading Dishwashers</Loader>
  ) : (
    <PageContainer>
      <header>
      <div className="d-flex justify-content-start">
        <NavigateBackButton onClick={() => navigate('/')}>&#8249;</NavigateBackButton>
        <H1 className="pt-4 pb-4">{title}</H1>
      </div>
      </header>
      <main>
        <section>
          {/* portrait on ipad */}
          <PortraitProductPage code={code}/>

          {/* landscape on ipad */}
          <LandscapeProductPage code={code}/>
        </section>
      </main>
    </PageContainer>
  );
};

ProductPage.propTypes = { productId: string };

export default ProductPage;
