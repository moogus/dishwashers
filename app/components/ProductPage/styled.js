import styled from 'styled-components';

export const H1 = styled.h1`
  font-size: 20px;
  font-weight: lighter;
  text-align: center;
  color: #444;
  margin: 0;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  @media (max-width: 768px) {
    margin: 0 10px;
  }
`;

export const H2 = styled.h2`
  font-size: 24px;
  font-weight: lighter;
  text-align: left;
  color: #444;
`;

export const H3 = styled.h2`
  font-size: 18px;
  text-align: left;
`;

export const Loader = styled.div`
  font-size: 24px;
  font-weight: normal;
  text-align: center;
  color: #333;
  margin-top: 45vh;
`;

export const PageContainer = styled.div`
  padding: 0;
  margin-bottom: 5px;
  margin-left: 0;
  margin-right:0;
`;

export const GridContainer = styled.div`
  background-color: #eee;
  padding: 2px;
`;

export const LineSpacer = styled.div`
  height: 5px;
  background-color: #eee;
`;

export const PortraitGridInnerContainer = styled.div`
  background-color: #fff;
  height: 100%;
  padding: 0;
`;

export const LanscapeGridInnerContainer = styled.div`
  background-color: #fff;
  height: 100%;
  padding: 0;
`;

export const TextMarginContainer = styled.div`
  margin-left: 5%;
  margin-right: 5%;
`;

export const NavigateBackButton = styled.button`
  color: #999;
  border: none;
  width: 5px;
  margin-left: 40px;
  margin-right: 40px;
  margin-top: -8px;
  font-size: 48px;
  font-weight: 100;
  &:focus {
      outline: 0 !important;
    }
  @media (max-width: 768px) {
    margin-left: 20px;
    margin-right: 20px;
  }
`;
