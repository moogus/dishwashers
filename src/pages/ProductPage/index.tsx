import React, { FunctionComponent } from "react";
import { useNavigate } from "@reach/router";

import { ProductTitle } from "../../components/ProductTitle";
import { PortraitProductSection } from "../../components/PortraitProductSection";
import { LandscapeProductSection } from "../../components/LanscapeProductSection";

import useLoadProduct from "../../hooks/product/load-product";

import { H1, NavigateBackButton, Loader, PageContainer } from "./styled";

type ProductPageProps = {
  path: string;
  productId?: string;
};

export const ProductPage: FunctionComponent<ProductPageProps> = ({
  productId,
}) => {
  const { loading } = useLoadProduct(productId);
  const navigate = useNavigate();

  return loading ? (
    <Loader>Loading Dishwashers</Loader>
  ) : (
    <PageContainer>
      <header>
        <div className="d-flex justify-content-start">
          <NavigateBackButton onClick={() => navigate("/")}>
            &#8249;
          </NavigateBackButton>
          <ProductTitle />
          <H1 className="pt-4 pb-4">{"title"}</H1>
        </div>
      </header>
      <main>
        <section>
          {/* portrait on ipad */}
          <PortraitProductSection />

          {/* landscape on ipad */}
          <LandscapeProductSection />
        </section>
      </main>
    </PageContainer>
  );
};
