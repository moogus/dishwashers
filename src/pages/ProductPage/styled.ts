import styled from 'styled-components';

export const H1 = styled.h1`
  font-size: 20px;
  font-weight: lighter;
  text-align: center;
  color: #444;
  margin: 0;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  @media (max-width: 768px) {
    margin: 0 10px;
  }
`;

export const Loader = styled.div`
  font-size: 24px;
  font-weight: normal;
  text-align: center;
  color: #333;
  margin-top: 45vh;
`;

export const PageContainer = styled.div`
  padding: 0;
  margin-bottom: 5px;
  margin-left: 0;
  margin-right:0;
`;

export const NavigateBackButton = styled.button`
  color: #999;
  border: none;
  width: 5px;
  margin-left: 40px;
  margin-right: 40px;
  margin-top: -8px;
  font-size: 48px;
  font-weight: 100;
  &:focus {
      outline: 0 !important;
    }
  @media (max-width: 768px) {
    margin-left: 20px;
    margin-right: 20px;
  }
`;
