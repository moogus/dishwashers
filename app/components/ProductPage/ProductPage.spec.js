import React from 'react';
import { render } from '@testing-library/react';

import ProductPage from 'components/ProductPage';
import useLoadProduct from 'hooks/product/load-product';
import useGetProduct from 'hooks/product/get-product';

jest.mock('hooks/product/load-product');
jest.mock('hooks/product/get-product');
jest.mock('@reach/router');

describe('<ProductPage />', () => {
  const renderContainer = ({ productId = 'foo' } = {}) => {
    const props = { productId };
    return render(<ProductPage {...props} />);
  };

  describe('when product api is loading', () => {
    beforeEach(() => {
      jest.clearAllMocks();
      useLoadProduct.mockReturnValue({ loading: true });
      useGetProduct.mockReturnValue({});
    });

    test('a loading message is displayed', () => {
      const { container } = renderContainer();
      expect(container).toMatchSnapshot();
    });
  });

  describe('when product api has loaded', () => {
    beforeEach(() => {
      jest.clearAllMocks();
      useLoadProduct.mockReturnValue({ loading: false });
      useGetProduct.mockReturnValue({
        title: 'foo',
        media: {
          altText: 'what the dickens',
          urls: ['foo'],
        },
        price: 'Â£1000',
        details: { productInformation: '<p>hiya</p>' },
        displaySpecialOffer: 'baz',
        additionalServices: { includedServices: ['stuff'] },
        code: '12345',
      });
    });

    test('a product page is rendered as expected', () => {
      const { container } = renderContainer();
      expect(container).toMatchSnapshot();
    });
  });
});
