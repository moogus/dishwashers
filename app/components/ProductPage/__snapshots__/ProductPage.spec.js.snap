// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductPage /> when product api has loaded a product page is rendered as expected 1`] = `
<div>
  <div
    class="styled__PageContainer-sc-7qvf1e-4 dmGTrW"
  >
    <header>
      <div
        class="d-flex justify-content-start"
      >
        <button
          class="styled__NavigateBackButton-sc-7qvf1e-10 glVeYK"
        >
          ‹
        </button>
        <h1
          class="styled__H1-sc-7qvf1e-0 czcIqx pt-4 pb-4"
        >
          foo
        </h1>
      </div>
    </header>
    <main>
      <section>
        <div
          class="d-none d-sm-block d-md-block d-lg-none"
        >
          <div
            class="styled__GridContainer-sc-7qvf1e-5 Rmvlk row"
          >
            <div
              class="styled__PortraitGridInnerContainer-sc-7qvf1e-7 fGdWSU col"
            >
              <div
                class="row"
              >
                <div
                  class="styled__FluidImageContainer-lcexfs-1 gBNScO col"
                >
                  <img
                    alt="what the dickens"
                    class="styled__FluidImage-lcexfs-0 gFFyIK d-flex justify-content-center"
                    data-testid="selected-image"
                    src="foo"
                  />
                </div>
              </div>
              <div
                class="row"
              >
                <div
                  class="col"
                >
                  <div
                    class="d-flex justify-content-center"
                    data-testid="image-select-buttons"
                  >
                    <button
                      class="styled__ActiveDotButton-lcexfs-2 dwkzOS"
                    >
                      •
                    </button>
                  </div>
                </div>
              </div>
              <div
                class="styled__LineSpacer-sc-7qvf1e-6 cUJxQd row pt1 pb1 border-top border-bottom"
              />
              <div
                class="styled__TextMarginContainer-om15ht-1 iGroMy row"
              >
                <div
                  class="col"
                >
                  <p
                    class="styled__PriceText-om15ht-0 Lzojg"
                  >
                    £1000
                  </p>
                  <p
                    class="styled__RedText-om15ht-2 kydASL"
                  >
                    Claim an extra 3 years guarantee via redemption
                  </p>
                  <p>
                    stuff
                  </p>
                </div>
              </div>
              <div
                class="styled__TextMarginContainer-sc-7qvf1e-9 fmfiij row"
              >
                <div
                  class="col"
                >
                  <h2
                    class="styled__H2-sc-7qvf1e-1 ifnKsI"
                  >
                    Product Information
                  </h2>
                </div>
              </div>
              <div
                class="styled__TextMarginContainer-sc-7qvf1e-9 fmfiij row"
              >
                <div
                  class="col"
                >
                  <h2
                    class="styled__H3-sc-7qvf1e-2 eYzICk"
                  >
                    Product code: 12345
                  </h2>
                </div>
              </div>
              <div
                class="styled__TextMarginContainer-sc-7qvf1e-9 fmfiij row"
              >
                <div
                  class="col"
                >
                  <span />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="d-none d-lg-block -xl-block"
        >
          <div
            class="styled__GridContainer-sc-7qvf1e-5 Rmvlk"
          >
            <div
              class="styled__LanscapeGridInnerContainer-sc-7qvf1e-8 caBZMz"
            >
              <div
                class="row"
              >
                <div
                  class="col-8  pr-0"
                >
                  <div
                    class="row"
                  >
                    <div
                      class="styled__FluidImageContainer-lcexfs-1 gBNScO col"
                    >
                      <img
                        alt="what the dickens"
                        class="styled__FluidImage-lcexfs-0 gFFyIK d-flex justify-content-center"
                        data-testid="selected-image"
                        src="foo"
                      />
                    </div>
                  </div>
                  <div
                    class="row"
                  >
                    <div
                      class="col"
                    >
                      <div
                        class="d-flex justify-content-center"
                        data-testid="image-select-buttons"
                      >
                        <button
                          class="styled__ActiveDotButton-lcexfs-2 dwkzOS"
                        >
                          •
                        </button>
                      </div>
                    </div>
                  </div>
                  <div
                    class="styled__TextMarginContainer-sc-7qvf1e-9 fmfiij row"
                  >
                    <div
                      class="col"
                    >
                      <h2
                        class="styled__H2-sc-7qvf1e-1 ifnKsI"
                      >
                        Product Information
                      </h2>
                    </div>
                  </div>
                  <div
                    class="styled__TextMarginContainer-sc-7qvf1e-9 fmfiij row"
                  >
                    <div
                      class="col "
                    >
                      <h2
                        class="styled__H3-sc-7qvf1e-2 eYzICk"
                      >
                        Product code: 12345
                      </h2>
                    </div>
                  </div>
                  <div
                    class="styled__TextMarginContainer-sc-7qvf1e-9 fmfiij row"
                  >
                    <div
                      class="col"
                    >
                      <span />
                    </div>
                  </div>
                </div>
                <div
                  class="col-4 border-left"
                >
                  <div
                    class="styled__TextMarginContainer-om15ht-1 iGroMy row"
                  >
                    <div
                      class="col"
                    >
                      <p
                        class="styled__PriceText-om15ht-0 Lzojg"
                      >
                        £1000
                      </p>
                      <p
                        class="styled__RedText-om15ht-2 kydASL"
                      >
                        Claim an extra 3 years guarantee via redemption
                      </p>
                      <p>
                        stuff
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
    </main>
  </div>
</div>
`;

exports[`<ProductPage /> when product api is loading a loading message is displayed 1`] = `
<div>
  <div
    class="styled__Loader-sc-7qvf1e-3 kZMDcq"
  >
    Loading Dishwashers
  </div>
</div>
`;
